src/main/java/au/org/ala/names/util/DwcaWriter.java

```diff
@@ -0,0 +1,413 @@
+package au.org.ala.names.util;
+
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import org.gbif.api.model.registry.Dataset;
+import org.gbif.dwc.terms.DcTerm;
+import org.gbif.dwc.terms.DwcTerm;
+import org.gbif.dwc.terms.Term;
+import org.gbif.dwca.io.Archive;
+import org.gbif.dwca.io.ArchiveField;
+import org.gbif.dwca.io.ArchiveFile;
+import org.gbif.dwca.io.MetaDescriptorWriter;
+import org.gbif.dwca.record.Record;
+import org.gbif.io.TabWriter;
+import org.gbif.registry.metadata.EMLWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * A more flexible DwcaWriter, based on the GBIF DwcaWriter.
+ * <p>
+ * This is interface compatible with the GBIF DwcaWriter but allows such things as
+ * non-linked extensions.
+ * </p>
+ *
+ * @see org.gbif.dwca.io.DwcaWriter
+ *
+ * @author Doug Palmer &lt;Doug.Palmer@csiro.au&gt;
+ * @copyright Copyright &copy; 2017 Atlas of Living Australia
+ */
+public class DwcaWriter {
```

Comment:
This class should extend AutoCloseable so that IDE's/static analysers can check for calls to the close method

Reply:
Cool